spikeinterface/core/core_tools.py

```diff
@@ -80,81 +81,67 @@ def write_python(path, dict):
             else:
                 f.write(str(k) + " = " + str(v) + "\n")
 
+class SIJsonEncoder(json.JSONEncoder):
+    """
+    An encoder used to encode Spike interface objects to json
+    """
+    def default(self, obj):
+        # Over-write behaviors for datetime object
+        if isinstance(obj, datetime.datetime):
+            return obj.isoformat()
+
+        # This should transforms integer, floats and bool to their python counterparts
+        if isinstance(obj, np.generic):
+            return obj.item()       
+                
+        if isinstance(obj, np.ndarray):
+            return obj.tolist()
+
+        if isinstance(obj, base.BaseExtractor):
+            return obj.to_dict()
+        
```

Comment:
Could we also add np.dtype ?

Reply:
Good idea, I guess they should be their strings, right?

Reply:
This is done. Check the last version of the tests.